crates/test-programs/src/bin/p3_http_outbound_request_content_length.rs

```diff
@@ -44,84 +44,93 @@ fn make_request() -> (
-                    .await
-                    .context("failed to finish body")?;
-                anyhow::Ok(res)
-            });
+            let (handle, (), ()) = join!(
+                async {
+                    let res = handler::handle(request)
+                        .await
+                        .context("failed to send request")?;
+                    anyhow::Ok(res)
+                },
+                async {
+                    let remaining = contents_tx.write_all(b"long enough".to_vec()).await;
+                    assert!(
+                        remaining.is_empty(),
+                        "{}",
+                        String::from_utf8_lossy(&remaining)
+                    );
+                    trailers_tx.write(Ok(None)).await.unwrap();
+                    drop(contents_tx);
+                },
+                async {
+                    transmit.await.unwrap();
+                },
+            );
             let res = handle.unwrap();
             drop(res);
-            transmit.expect("failed to transmit request");
         }
 
+        println!("writing too little");
         {
             let (request, mut contents_tx, trailers_tx, transmit) = make_request();
-            let (transmit, handle) = join!(async { transmit.await }, async {
-                let res = handler::handle(request)
-                    .await
-                    .context("failed to send request")?;
-                println!("writing too little");
-                let remaining = contents_tx.write_all(b"msg".to_vec()).await;
-                assert!(
-                    remaining.is_empty(),
-                    "{}",
-                    String::from_utf8_lossy(&remaining)
-                );
-                drop(contents_tx);
-                trailers_tx
-                    .write(Ok(None))
-                    .await
-                    .context("failed to finish body")?;
-                anyhow::Ok(res)
-            });
-            let res = handle.unwrap();
-            drop(res);
-            let err = transmit.expect_err("request transmission should have failed");
-            assert!(
-                matches!(err, ErrorCode::HttpRequestBodySize(Some(3))),
-                "unexpected error: {err:#?}"
+            let (handle, (), ()) = join!(
+                async { handler::handle(request).await },
+                async {
+                    transmit.await.unwrap();
+                },
+                async {
+                    let remaining = contents_tx.write_all(b"msg".to_vec()).await;
+                    assert!(
+                        remaining.is_empty(),
+                        "{}",
+                        String::from_utf8_lossy(&remaining)
+                    );
+                    drop(contents_tx);
+                    trailers_tx.write(Ok(None)).await.unwrap();
+                },
             );
+            // FIXME(#11631) should have a more precise error like the one
```

Comment:
There's currently no `Content-Length` header handling, so I would not modify this test until that's implemented in the host.